airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -1074,15 +1086,19 @@ def create_concurrent_cursor_from_datetime_based_cursor(
                     clamping_strategy = WeekClampingStrategy(weekday)
                     end_date_provider = ClampingEndProvider(
                         WeekClampingStrategy(weekday, is_ceiling=False),
-                        end_date_provider,
-                        granularity=timedelta(days=1),
+                        end_date_provider,  # type: ignore  # Having issues w/ inspection for GapType and CursorValueType as shown in existing tests. Confirmed functionality is working in practice
+                        granularity=cursor_granularity or timedelta(days=1),
```

Comment:
We don't have a case here but I would assume it should be only `cursor_granularity` here. The downside I see with fallbacking to `timedelta(days=1)` is that if we have a clamping WeekClampingStrategy(Weekday.Monday) and we are on Monday, the sync will be missing the last day if `cursor_granularity` is not defined